### ISSUE TITLE
Can the camera be prepared in other thread to avoid UI glitches while opening?

### ISSUE THREAD

**Issue by usmanrana-ilsa** [natario1/CameraView#1158]
### Version used
2.7.2

**Comment by anggit97** [comment]
@usmanrana-ilsa do you have solve this problem?

**Comment by usmanrana07** [comment]
In my case I was doing some other operations too during swipe on page. Holding them unless camera fragment is resumed improved the performance. Otherwise camera is preparing in main thread and that would cause some glitch. Almost similar behavior is observed in Instagram camera as well.

**Comment by anggit97** [comment]
im using view pager with 3 fragments. when page is swiped, i'm not doing any  operation. but its still show some glitch (showing overlay gray, before camera preview is loaded). do you experience it?

**Comment by usmanrana07** [comment]
yes, CameraView isn't resuming from last paused preview, but shows shuttering effect while resume again

**Comment by tashaerxing** [comment]
How to solve the problem of slow focus when opening？
